### PR TITLE
Create CurrentUserTests.cs

### LINKED ISSUE #382: Add tests for CurrentUserService
Currently, there are no unit tests for `CurrentUserService`. We need to implement unit tests for this service.

### PR DESCRIPTION
**Pull request by Pasha1968**
Created CurrentUserTests.cs - tests for CurrentUserService.cs

### CODE REVIEW

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            var currentuserService = new CurrentUserService(_httpContextAccessorMock.Object);
+            Assert.Throws<UnauthorizedAccessException>(() => currentuserService.AccountId);
+        }
+
+        [Fact]
+        public void getAccountId()
+        {
+            var _httpContextAccessorMock = new Mock<IHttpContextAccessor>();
+            var httpContext = new DefaultHttpContext();
+            var mockIdentity = new Mock<GenericIdentity>("Vasya");
```
Comment (gingerbread-monster) [inline CharlieBackend.Api.UnitTes/CurrentUserTests.cs]: I think you can move this repetitive boilerplate code from each test method into `CurrentUserTests` class constructor, can't you?

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                .Returns(new Claim[] {
+                    new Claim("Email", "Hello@mail.com") 
+                });
+            httpContext.User = new ClaimsPrincipal(mockIdentity.Object);
+            _httpContextAccessorMock.Setup(req => req.HttpContext).Returns(httpContext);
+
+            var currentuserService = new CurrentUserService(_httpContextAccessorMock.Object);
+            var Email = currentuserService.Email;
+
+            Assert.Equal("Hello@mail.com", Email);
```
Comment (gingerbread-monster) [inline CharlieBackend.Api.UnitTes/CurrentUserTests.cs]: Also I think we're supposed to use **FluentAssertions** for tests. You can view `MentorServiceTests` for reference.

e.g.:
https://github.com/ita-social-projects/WhatBackend/blob/899f397be4e51f81fc156f059c3ced0a0a60d669/CharlieBackend.Api.UnitTes/MentorServiceTests.cs#L65

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        [Fact]
+        public void getEmail()
+        {
+
+            _mockIdentity.Setup(i => i.Claims)
+                .Returns(new Claim[] {
+                    new Claim("Email", "Hello@mail.com") 
+                });
+            _httpContext.User = new ClaimsPrincipal(_mockIdentity.Object);
+            _httpContextAccessorMock.Setup(req => req.HttpContext).Returns(_httpContext);
```
Comment (gingerbread-monster) [inline CharlieBackend.Api.UnitTes/CurrentUserTests.cs]: Aren't these 2 lines can be moved from each method into constructor as well? :)

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        }
+
+        [Fact]
+        public void ThrowErrorIfNotLoggined()
+        {
+            _httpContextAccessorMock.Setup(req => req.HttpContext)
+                .Returns(new DefaultHttpContext());
+
+            Func<long> getId = () => _currentuserService.AccountId;
+            Action act = () => getId();
```
Comment (gingerbread-monster) [inline CharlieBackend.Api.UnitTes/CurrentUserTests.cs]: Why add redundant delegates? Wouldn't it already work with the first one?

```suggestion
            Func<long> act = () => _currentuserService.AccountId;
```

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        public void getAccountId()
+        {
+            _mockIdentity.Setup(i => i.Claims)
+                .Returns(new Claim[]{
+                    new Claim("AccountId", "123") 
+                });
+            _httpContext.User = new ClaimsPrincipal(_mockIdentity.Object);
+            _httpContextAccessorMock.Setup(req => req.HttpContext).Returns(_httpContext);
+
+            _currentuserService.AccountId.Should().Be(123);
```
Comment (gingerbread-monster) [inline CharlieBackend.Api.UnitTes/CurrentUserTests.cs]: When introducing expected values as constants, what I would suggest is declaring them in a single place (at least in the `Arrange` section of the test). This way you're decreasing the possibility of making a mistake and making code more "fluent".
Something like this:

```cs
            // Arrange
            long expectedAccountId = 123;
            
            _mockIdentity.Setup(i => i.Claims)
                .Returns(new Claim[]{
                    new Claim("AccountId", expectedAccountId.ToString()) 
                });

            // Act & Assert
            _currentuserService.AccountId.Should().Be(expectedAccountId);
```

This comment also applies to other methods & constants of yours.

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs (hunk 1 above)
Comment (gingerbread-monster) [reply CharlieBackend.Api.UnitTes/CurrentUserTests.cs]: Partially solved in https://github.com/ita-social-projects/WhatBackend/pull/504/commits/e24ea4bfff6c941a77091f4f006367542332ab6b

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs (hunk 2 above)
Comment (gingerbread-monster) [reply CharlieBackend.Api.UnitTes/CurrentUserTests.cs]: Solved in https://github.com/ita-social-projects/WhatBackend/pull/504/commits/e24ea4bfff6c941a77091f4f006367542332ab6b

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+            _httpContextAccessorMock = new Mock<IHttpContextAccessor>();
+            _httpContext = new DefaultHttpContext();
+            _mockIdentity = new Mock<GenericIdentity>("Test");
+            _currentuserService = new CurrentUserService(_httpContextAccessorMock.Object);
+            _httpContext.User = new ClaimsPrincipal(_mockIdentity.Object);
+            _httpContextAccessorMock.Setup(req => req.HttpContext).Returns(_httpContext);
+        }
+
+        [Fact]
+        public void ThrowErrorIfNotLoggined()
```
Comment (andrii-developer) [inline CharlieBackend.Api.UnitTes/CurrentUserTests.cs]: Please try to follow [best practices](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices?ranMID=46131&ranEAID=a1LgFw09t88&ranSiteID=a1LgFw09t88-PBLgRQP1_o9AbEiA2hJ8zg&epi=a1LgFw09t88-PBLgRQP1_o9AbEiA2hJ8zg&irgwc=1&OCID=AID2000142_aff_7806_1243925&tduid=%28ir__nnr3tb2a2okfqj1hkk0sohz3132xusa63bxexe1w00%29%287806%29%281243925%29%28a1LgFw09t88-PBLgRQP1_o9AbEiA2hJ8zg%29%28%29&irclickid=_nnr3tb2a2okfqj1hkk0sohz3132xusa63bxexe1w00#naming-your-tests) for naming of your tests. 
Please rename test methods like this one:
```suggestion
        public void AccountId_UserNotLoggedIn_ThrowsUnauthorizedAccessException()
```

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+            // Act
+            Func<long> act = () => _currentuserService.AccountId;
+
+            //Assert
+            act.Should().Throw<UnauthorizedAccessException>();
+        }
+
+        [Fact]
+        public void getAccountId()
```
Comment (andrii-developer) [inline CharlieBackend.Api.UnitTes/CurrentUserTests.cs]: Please follow [C# coding conventions](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions). 
When naming public members of types, such as fields, properties, events, methods, and local functions, use pascal casing.

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+            string expectedEmail = "Hello@mail.com";
+
+            _mockIdentity.Setup(i => i.Claims)
+                .Returns(new Claim[] {
+                    new Claim("Email", expectedEmail) 
+                });
+
+            // Act & Assert
+            _currentuserService.Email.Should().BeEquivalentTo(expectedEmail);
+        }
```
Comment (andrii-developer) [inline CharlieBackend.Api.UnitTes/CurrentUserTests.cs:87]: Please add one blank line between methods.

**Review by andrii-developer — CHANGES_REQUESTED** [review]
Waiting for the author to address comments.

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs [hunk 9]
```diff
@@ -42,56 +44,57 @@ public void ThrowErrorIfNotLoggined()
...
                 });
-            _httpContext.User = new ClaimsPrincipal(_mockIdentity.Object);
-            _httpContextAccessorMock.Setup(req => req.HttpContext).Returns(_httpContext);
 
-            _currentuserService.AccountId.Should().Be(123);
+            // Act & Assert
+            _currentuserService.AccountId.Should().Be(expectedAccountId);
         }
         [Fact]
         public void getEntityId()
```
Comment (gingerbread-monster) [inline CharlieBackend.Api.UnitTes/CurrentUserTests.cs]: You're the first person I see to completely ignore the coding style of the rest of the project  and use _camelCase_ methods naming. 
Come on, it's not JavaScript. We use _PascalCase_ naming in C#.

Even better if you'd follow [the unit test naming convention](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#naming-your-tests). You can also look at other unit tests for reference.

Comment (gingerbread-monster) [reply CharlieBackend.Api.UnitTes/CurrentUserTests.cs]: Duplicate of https://github.com/ita-social-projects/WhatBackend/pull/504#discussion_r646455509

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs (hunk 5 above)
Comment (gingerbread-monster) [reply CharlieBackend.Api.UnitTes/CurrentUserTests.cs]: Solved in https://github.com/ita-social-projects/WhatBackend/pull/504/commits/599f172a458febfc3f38a1fd66639d04ac39d823

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs (hunk 4 above)
Comment (gingerbread-monster) [reply CharlieBackend.Api.UnitTes/CurrentUserTests.cs]: Solved in https://github.com/ita-social-projects/WhatBackend/pull/504/commits/599f172a458febfc3f38a1fd66639d04ac39d823

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs (hunk 3 above)
Comment (gingerbread-monster) [reply CharlieBackend.Api.UnitTes/CurrentUserTests.cs]: Solved in https://github.com/ita-social-projects/WhatBackend/pull/504/commits/599f172a458febfc3f38a1fd66639d04ac39d823

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+                .Returns(new Claim[] {
+                    new Claim("Email", expectedEmail) 
+                });
+
+            // Act & Assert
+            _currentuserService.Email.Should().BeEquivalentTo(expectedEmail);
+        }
+
+        [Fact]
+
```
Comment (andrii-developer) [inline CharlieBackend.Api.UnitTes/CurrentUserTests.cs]: Remove empty lines between [Fact] attribute and methods.

File: CharlieBackend.Api.UnitTes/CurrentUserTests.cs [hunk 11]
```diff
@@ -0,0 +1,103 @@
+﻿using Moq;
+using Xunit;
+using System;
+using System.Security.Claims;
+using System.Security.Principal;
+using Microsoft.AspNetCore.Http;
+using CharlieBackend.Core.Entities;
+using CharlieBackend.Business.Services;
+using FluentAssertions;
+
```
Comment (andrii-developer) [inline CharlieBackend.Api.UnitTes/CurrentUserTests.cs:10]: Remove this empty line.

**Review by andrii-developer — APPROVED** [review]

**Review by gingerbread-monster — APPROVED** [review]

**Review by DemetriosKoras — APPROVED** [review]

**Pull request merged**